juicy-html.html

```diff
@@ -44,47 +44,73 @@
+                            return model;
                         }
                     },
-                    get: function () {
-                        return this.getAttribute('href');
-                    }
-                },
-                'html': {
-                    set: function (newValue) {
-                        if (nullOrUndefined(newValue)) {
-                            this.removeAttribute('html');
-                        } else {
-                            this.setAttribute('html', newValue);
+                    'href': {
+                        set: function (newValue) {
+                            if (nullOrUndefined(newValue)) {
+                                this.removeAttribute('href');
+                            } else {
+                                this.setAttribute('href', newValue);
+                            }
+                        },
+                        get: function () {
+                            return this.getAttribute('href');
                         }
                     },
-                    get: function () {
-                        return this.getAttribute('html');
+                    'html': {
+                        set: function (newValue) {
+                            if (nullOrUndefined(newValue)) {
+                                this.removeAttribute('html');
+                            } else {
+                                this.setAttribute('html', newValue);
+                            }
+                        },
+                        get: function () {
+                            return this.getAttribute('html');
+                        }
                     }
-                }
-            });
+                });
+            }
+            /** Autostamp on connected */
+            connectedCallback(){
+                this.isAttached = true;
+                // Workaround for lack of inertness as for `<tempalte is="juicy-html">`.
+                // Bellow is cheaper and simmpler than attaching shadow with display none
+                // it does not give exactly the same behavior, but hopefully nobody would like to make it block.
+                this.style.display = 'none';
+                // autostamp
+                // TODO(tomalec): read pre-upgrade properties in cheap manner
+                this.hasAttribute('href') && this.attributeChangedCallback('href', null, this.getAttribute('href'));
+                this.hasAttribute('html') && this.attributeChangedCallback('html', null, this.getAttribute('html'));
+                // translate legacy `content` attribute
+                this.hasAttribute('content') && this.attributeChangedCallback('content', null, this.getAttribute('content'));
+            }
+            disconnectedCallback(){
+                this.isAttached = false;
+                this.clear();
+            }
         };
+        var JuicyHTMLPrototype = JuicyHTMLElement.prototype;
```

Comment:
Still use prototypes? `clear`, `isAttached` and `attributeChangedCallback` etc.., could fall perfectly under the class. And would look super neat if were ES6 consice methods: 
```js
{ // the class
    skipStampingPendingFile() {

    }
}
```

Reply:
AS above, I made it that way, mostly, to reduce the number of changes for easier diffing and history tracking. I'll change it to class methods.

Reply:
Well thanks, reviewing this was a breeze.

Reply:
Another round won't be, as I moved everything to the class ;)